### PR TITLE
tls: introduce Bundle() and Bundler.Bundle()

### PR DESCRIPTION
**Pull request by amery**
to assemble [tls.Certificate]s

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `Bundler` feature for managing and bundling TLS certificates and keys.
	- Added methods for bundling certificates and resetting configurations.
	- Enhanced validation checks for certificate and key pairs.
- **Chores**
	- Updated dependency to improve compatibility and performance.

These updates improve the flexibility and reliability of TLS certificate management within the application.

### CODE REVIEW

**Comment by amery** [comment]
apply grammar fixes caught by @coderabbitai

File: tls/bundler.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+import (
+	"crypto/tls"
+	"crypto/x509"
+
+	"darvaza.org/core"
+	"darvaza.org/x/tls/x509utils"
+	"darvaza.org/x/tls/x509utils/certpool"
+)
+
+// Bundler uses two CertPools to bundler keys and certificates
```
Comment (karasz) [inline tls/bundler.go:12]: to bundle, not to bundler.

**Review by karasz — CHANGES_REQUESTED** [review]

**Comment by amery** [comment]
fixed grammar issue spotted by @karasz

**Comment by amery** [comment]
rebased, fixed logic error spotted by @coderabbitai, and refactored to take advantage of #112 (now merged)

File: tls/bundler.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+	switch {
+	case s.Inter == nil:
+		s.opts.Intermediates = nil
+	case s.opts.Intermediates == nil:
+		s.opts.Intermediates = s.Inter.Export()
+	}
+
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (amery) [reply tls/bundler.go]: Mutex added

File: tls/bundler.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+// Reset drops any cached information.
+func (s *Bundler) Reset() {
+	s.opts.Roots = nil
+	s.opts.Intermediates = nil
+}
```
[the remark this answers is not in the document]
Comment (amery) [reply tls/bundler.go]: Mutex added

File: tls/bundler.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+	// Root Certificates. Defaults to system's.
+	Roots x509utils.CertPool
+	// Intermediate Certificates.
+	Inter x509utils.CertPool
+	// Quality comparison function. Defaults to shorter-chain.
+	Less func(a, b []*x509.Certificate) bool
+
+	mu   sync.Mutex
+	opts x509.VerifyOptions
+}
```
[the remark this answers is not in the document]
Comment (karasz) [reply tls/bundler.go:24]: imvho this is not really necessary here.

**Review by karasz — APPROVED** [review]

**Comment by amery** [comment]
`darvaza.org/x/tls@v0.2.6` tagged

**Pull request merged**